### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by cameronbraid** [vmware-tanzu/carvel-kapp#73]
kapp version

`Client Version: 0.19.0`

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xb8 pc=0xf60a77]

goroutine 1 [running]:
github.com/k14s/kapp/pkg/kapp/resources.UniqueResourceKey.String(0x0, 0x0, 0x0, 0x0, 0xb, 0x1cf28c0)
        github.com/k14s/kapp@/pkg/kapp/resources/unique_resources.go:23 +0x37
github.com/k14s/kapp/pkg/kapp/clusterapply.ConvergedResource.findParentAndAssociatedRes(0x0, 0x0, 0x0, 0x0, 0x0, 0xc00010dec0, 0xb, 0xb, 0x0, 0x0, ...)
        github.com/k14s/kapp@/pkg/kapp/clusterapply/converged_resource.go:111 +0x5f
github.com/k14s/kapp/pkg/kapp/clusterapply.ConvergedResource.IsDoneApplying(0x0, 0x0, 0x0, 0x0, 0x0, 0xc00010dec0, 0xb, 0xb, 0x0, 0x0, ...)
        github.com/k14s/kapp@/pkg/kapp/clusterapply/converged_resource.go:33 +0x6a
github.com/k14s/kapp/pkg/kapp/clusterapply.NewValueResourceConverged(0x0, 0x0, 0x0, 0x2, 0xc0003dbdc8, 0xc000867100)
        github.com/k14s/kapp@/pkg/kapp/clusterapply/changes_view.go:227 +0xd2
github.com/k14s/kapp/pkg/kapp/clusterapply.(*ChangesView).Print(0xc0000be300, 0x14a80c0, 0xc0002dd600)
        github.com/k14s/kapp@/pkg/kapp/clusterapply/changes_view.go:110 +0x10eb
github.com/k14s/kapp/pkg/kapp/clusterapply.(*ChangeSetView).Print(0xc000538e30, 0x14a80c0, 0xc0002dd600)
        github.com/k14s/kapp@/pkg/kapp/clusterapply/change_set_view.go:41 +0x3c8
github.com/k14s/kapp/pkg/kapp/cmd/app.(*DeployOptions).calculateAndPresentChanges(0xc0002aab40, 0xc0005949c0, 0x3, 0x4, 0xc0000e6600, 0x9, 0x10, 0xc0004aca80, 0x2, 0x2, ...)
        github.com/k14s/kapp@/pkg/kapp/cmd/app/deploy.go:317 +0xa07
github.com/k14s/kapp/pkg/kapp/cmd/app.(*DeployOptions).Run(0xc0002aab40, 0x0, 0x0)
        github.com/k14s/kapp@/pkg/kapp/cmd/app/deploy.go:136 +0x7a8
github.com/k14s/kapp/pkg/kapp/cmd/app.NewDeployCmd.func1(0xc0003fef00, 0xc0002f18f0, 0x0, 0x7, 0xc00018a420, 0xc000113be0)
        github.com/k14s/kapp@/pkg/kapp/cmd/app/deploy.go:48 +0x2a
github.com/k14s/kapp/pkg/kapp/cmd.NewKappCmd.func1.1(0xc0003fef00, 0xc0002f18f0, 0x0, 0x7, 0x0, 0x0)
        github.com/k14s/kapp@/pkg/kapp/cmd/kapp.go:122 +0xcc
github.com/cppforlife/cobrautil.ReconfigureLeafCmd.func1(0xc0003fef00, 0xc0002f18f0, 0x0, 0x7, 0x0, 0x0)
        github.com/cppforlife/cobrautil@v0.0.0-20200108171912-b289811fd2b9/misc.go:67 +0x135
github.com/cppforlife/cobrautil.WrapRunEForCmd.func1.1(0xc0003fef00, 0xc0002f18f0, 0x0, 0x7, 0x0, 0x0)
        github.com/cppforlife/cobrautil@v0.0.0-20200108171912-b289811fd2b9/misc.go:25 +0xaf
github.com/cppforlife/cobrautil.WrapRunEForCmd.func1.1(0xc0003fef00, 0xc0002f18f0, 0x0, 0x7, 0x0, 0x0)
        github.com/cppforlife/cobrautil@v0.0.0-20200108171912-b289811fd2b9/misc.go:25 +0xaf
github.com/spf13/cobra.(*Command).execute(0xc0003fef00, 0xc0002f1880, 0x7, 0x7, 0xc0003fef00, 0xc0002f1880)
        github.com/spf13/cobra@v0.0.3/command.go:762 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc0003fe500, 0xc0003fe500, 0xc0002dd600, 0x43b9ea)
        github.com/spf13/cobra@v0.0.3/command.go:852 +0x2ea
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v0.0.3/command.go:800
main.main()
        github.com/k14s/kapp@/cmd/kapp/kapp.go:27 +0x16f```

**Comment by cppforlife** [comment]
interesting... @cameronbraid are you using `kapp.k14s.io/versioned` annotation?

**Comment by cameronbraid** [comment]
yeah, this came from me applying `kapp.k14s.io/versioned` to a deployment configuration by mistake

**Comment by lnhrdt** [comment]
Hey @cppforlife, I encountered this error today, also while using the `kapp.k14s.io/versioned`, but on a Job. I only encountered the error when the job failed (in an "Error" state) and then I tried to deploy the job again.

Is there any way to use the `kapp.k14s.io/versioned` feature without encountering this error?

(kapp version 0.34.0)

**Comment by cppforlife** [comment]
@lnhrdt i couldnt repro this error before. if you can provide steps you followed with some example config that would be greatly appreciated.

**Comment by praveenrewar** [comment]
Related to #491. Fix available as part of kapp v0.47.0+
